### PR TITLE
[v2] Merge main on Oct. 18

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Merges `main` branch into the `feature-arch-v2` and fixes all conflicts and issues. There was a lot of conflicts, I spent considerable amount of time resolving them and ensures that all new fixes are included into v2 as long as they fit into the model that v2 operates in.

## Why is it important?

Preparation for `feature-arch-v2` to be merged into `main` on Oct. 26th.

### CODE REVIEW

**Comment by cmacknz** [comment]
We will need to backport this change to the Heartbeat v1 spec file: https://github.com/elastic/elastic-agent/pull/1491

We just need to have it only support the Elasticsearch output.

**Comment by blakerouse** [comment]
@cmacknz I updated the new v2 specification file to only have elasticsearch as a supported output for heartbeat.

File: internal/pkg/agent/application/application.go [hunk 1]
```diff
@@ -111,7 +113,7 @@ func New(
 		}
 	}
 
-	composable, err := composable.New(log, rawConfig)
```
Comment (michalpristas) [inline internal/pkg/agent/application/application.go:114]: seems like we should address changes from #848 and #686  here as well

Comment (blakerouse) [reply internal/pkg/agent/application/application.go:114]: yeah hopefully I got all that in the merge, I ensured all the reload interfaces are present

**Review by michalpristas — APPROVED** [review]
merge looks good, we will retest for regressions and address issues as they pop up. but important things seem to be included

**Pull request merged**